freeipa/src/main/java/com/sequenceiq/freeipa/service/AltusMachineUserService.java

```diff
@@ -113,14 +142,34 @@ public DataBusCredential getOrCreateDataBusCredentialIfNeeded(Stack stack) throw
         } else {
             LOGGER.debug("Databus credential does not exist for the stack, it will be created ...");
         }
-        Optional<AltusCredential> altusCredential = createMachineUserWithAccessKeys(stack, telemetry);
+        Optional<AltusCredential> altusCredential = createDatabusMachineUserWithAccessKeys(stack, telemetry);
         return storeDataBusCredential(altusCredential, stack);
     }
 
+    public MonitoringCredential getOrCreateMonitoringCredentialIfNeeded(Stack stack) throws IOException {
+        LOGGER.debug("Get or create databus credential for stack");
+        Telemetry telemetry = stack.getTelemetry();
+        if (stack.getMonitoringCredential() != null) {
+            LOGGER.debug("Monitoring credential has been found for the stack");
+            MonitoringCredential monitoringCredential = new Json(stack.getMonitoringCredential()).get(MonitoringCredential.class);
+            if (isCredentialExists(telemetry, monitoringCredential.getMachineUserName(), monitoringCredential.getAccessKey(), stack)) {
+                LOGGER.debug("Monitoring credential exists both in the stack and on UMS side");
+                return monitoringCredential;
```

Comment:
could you refactor so there won't be a return in the middle of the method? I would suggest splitting this into methods